projects/rocprofiler-systems/source/lib/rocprof-sys/library/rocprofiler-sdk.cpp

```diff
@@ -1698,6 +1717,119 @@ tool_tracing_buffered(rocprofiler_context_id_t /*context*/,
+                                           "] Thread ", record->thread_id);
+                    cache_category<category::rocm_scratch_memory>();
+                    cache_add_thread_info(record->thread_id);
+                    cache_add_track(track_name.c_str(), record->thread_id);
+                    cache_scratch_memory(record, _stream_id);
+                }
+
+                if(get_use_timemory())
+                {
+                    const auto& _tinfo = thread_info::get(record->thread_id, SystemTID);
+                    auto        _tid   = _tinfo->index_data->sequent_value;
+
+                    auto _bundle = kernel_dispatch_bundle_t{ _name };
+
+                    _bundle.push(_tid).start().stop();
+                    _bundle.get([_beg_ns, _end_ns](tim::component::wall_clock* _wc) {
+                        _wc->set_value(_end_ns - _beg_ns);
+                        _wc->set_accum(_end_ns - _beg_ns);
+                    });
+                    _bundle.pop();
+                }
+
+                if(get_use_perfetto())
+                {
+// Scratch memory samples from SDK versions prior to 7.0.2 do not include
+// allocation_size field, so counter tracks are not needed
+#if(ROCPROFSYS_USE_ROCM > 0 && ROCPROFSYS_ROCM_VERSION >= 70002)
+                    using counter_track = perfetto_counter_track<
+                        rocprofiler_buffer_tracing_scratch_memory_record_t>;
+
+                    if(!counter_track::exists(device_id))
+                    {
+                        auto track_name = JOIN("", "GPU Scratch Memory [", device_id,
+                                               "] (S) Alloc Thread ", record->thread_id);
+                        counter_track::emplace(device_id, track_name, "bytes");
+                    }
+
+                    if(record->operation == ROCPROFILER_SCRATCH_MEMORY_ALLOC)
+                    {
+                        TRACE_COUNTER("rocm_scratch_memory",
+                                      counter_track::at(device_id, 0), _beg_ns,
+                                      record->allocation_size);
+                    }
+#endif
+                    auto add_perfetto_annotations = [&](::perfetto::EventContext ctx) {
+                        if(config::get_perfetto_annotations())
+                        {
+                            tracing::add_perfetto_annotation(ctx, "begin_ns", _beg_ns);
+                            tracing::add_perfetto_annotation(ctx, "end_ns", _end_ns);
+                            tracing::add_perfetto_annotation(ctx, "corr_id", _corr_id);
+                            tracing::add_perfetto_annotation(ctx, "stream_id",
+                                                             _stream_id);
+                        }
+                    };
+
+                    if(_group_by_queue)
+                    {
+                        auto track_name = [&]() {
+                            return JOIN("", "GPU Scratch Memory (S) Events Thread ",
```

Comment:
I actually meant something like this. So that the thread IDD is the same as for memory_copy
```suggestion
                            const auto& t_info = thread_info::get(record->thread_id, SystemTID);
                            auto             thread_id   = t_info->index_data->sequent_value;
                            return JOIN("", "GPU Scratch Memory (S) Events Thread ", thread_id
```

Reply:
Switched to sequent thread ID in both legacy and cached perfetto implementation